zio-http/src/main/scala/zhttp/http/middleware/Metrics.scala

```diff
@@ -0,0 +1,102 @@
+package zhttp.http.middleware
+
+import zhttp.http.{Http, Middleware, Request, Response}
+import zio._
+import zio.metrics.{Metric, MetricKeyType, MetricLabel}
+
+private[zhttp] trait Metrics {
+  def metrics[R, E](
+    mapper: Metrics.Mapper = Map.empty,
+    boundaries: MetricKeyType.Histogram.Boundaries = Metrics.defaultBoundaries,
+  ): HttpMiddleware[R with Metrics.InFlightRequests, E] =
+    new Middleware[R with Metrics.InFlightRequests, E, Request, Response, Request, Response] {
```

Comment:
I think there is quite a lot of logic that's got nothing to do with a middleware. 
How about we create a `Metrics` service that does all this stuff we test that directly instead of doing it via the middleware API.

That way we will later get flexibility to also benchmark the service and see how it performs under stress.

Reply:
I don't really see what that service would do – all helpers essentially extract different metadata about the request or the response to appropriately tag the metrics with. We can definitely move that to its own module if you think it helps readability. 👍